tools/PsesPsClient/PsesPsClient.psm1

```diff
@@ -0,0 +1,314 @@
+        [string]
+        $LogLevel = 'Diagnostic',
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $SessionDetailsPath,
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [string[]]
+        $FeatureFlags = @('PSReadLine'),
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $HostName = 'PSES Test Host',
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $HostProfileId = 'TestHost',
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [version]
+        $HostVersion = '1.99',
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [string[]]
+        $AdditionalModules = @('PowerShellEditorServices.VSCode'),
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $BundledModulesPath,
+
+        [Parameter()]
+        [switch]
+        $EnableConsoleRepl
+    )
+
+    $EditorServicesPath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($EditorServicesPath)
+
+    $instanceId = Get-RandomHexString
+
+    $tempDir = [System.IO.Path]::GetTempPath()
+
+    if (-not $LogPath)
+    {
+        $LogPath = Join-Path $tempDir "pseslogs_$instanceId.log"
+    }
+
+    if (-not $SessionDetailsPath)
+    {
+        $SessionDetailsPath = Join-Path $tempDir "psessession_$instanceId.log"
+    }
+
+    if (-not $BundledModulesPath)
+    {
+        $BundledModulesPath = $script:PsesBundledModulesDir
+    }
+
+    $editorServicesOptions = @{
+        LogPath = $LogPath
+        LogLevel = $LogLevel
+        SessionDetailsPath = $SessionDetailsPath
+        FeatureFlags = $FeatureFlags
+        HostName = $HostName
+        HostProfileId = $HostProfileId
+        HostVersion = $HostVersion
+        AdditionalModules = $AdditionalModules
+        BundledModulesPath = $BundledModulesPath
+        EnableConsoleRepl = $EnableConsoleRepl
+    }
+
+    $startPsesCommand = Unsplat -Prefix "& '$EditorServicesPath'" -SplatParams $editorServicesOptions
+
+    $pwshPath = (Get-Process -Id $PID).Path
+
+    if (-not $PSCmdlet.ShouldProcess("& '$pwshPath' -Command '$startPsesCommand'"))
+    {
+        return
+    }
+
+    $serverProcess = Start-Process -PassThru -FilePath $pwshPath -ArgumentList @(
+        '-NoLogo',
+        '-NoProfile',
+        '-NoExit',
+        '-Command',
+        $startPsesCommand
+    )
+
+    $sessionPath = $editorServicesOptions.SessionDetailsPath
+
+    $i = 0
+    while (-not (Test-Path $sessionPath))
+    {
+        if ($i -ge 10)
+        {
+            throw "No session file found - server failed to start"
+        }
+
+        Start-Sleep 1
+        $null = $i++
+    }
+
+    return [PsesServerInfo]@{
+        PsesProcess = $serverProcess
+        SessionDetails = Get-Content -Raw $editorServicesOptions.SessionDetailsPath | ConvertFrom-Json
+        StartupOptions = $editorServicesOptions
+    }
+}
+
+function Connect-NamedPipe
```

Comment:
I'd rather call this like... `Connect-PsesServer` which returns a `PsesPsClient` that you pass into a `Send-Lsp*Request` and Receive-LspResponse` I feel as if it will make writing tests easier by not having the pipe-specific knowledge.